docs/modules/data-structures/pages/working-with-external-data.adoc

```diff
@@ -250,6 +250,57 @@ include::ROOT:example$/dds/map/PersonEntryStore.java[tag=personms]
 
 TIP: link:https://github.com/hazelcast/hazelcast-code-samples/tree/master/distributed-map/mapstore/src/main/java[See more MapStore/MapLoader code samples].
 
+== Offloading MapStore and MapLoader Operations
+
+Offloading `MapStore` and `MapLoader` operations allow these operations to work asynchronously, meaning they do not block each other.
```

Comment:
@ahmetmircik WDYT of this?

Reply:
>To improve cluster throughput, most MapStore operations do not block partition threads.

Not most but all MapStore operations are async by default. Only `MapLoaderLifecycleSupport` is sync but it runs only one time during the creation of IMap.

Reply:
> > MapStore operations that communicate with the external data source are handled asynchronously, using an entry processor.
> 
> I didn't get the relation between EP and async map-store calls. Async map-store calls are done independently, EP is not required.

It's an executor instead?

Reply:
> > To improve cluster throughput, most MapStore operations do not block partition threads.
> 
> Not most but all MapStore operations are async by default. Only `MapLoaderLifecycleSupport` is sync but it runs only one time during the creation of IMap.

Since `MapLoaderLifecycleSupport` is sync, isn't it incorrect to say all?

Reply:
>Since MapLoaderLifecycleSupport is sync, isn't it incorrect to say all?

No, it is a separate interface and implementing it is not required, it is up-to-user. If user is implemented it, it is blocking.

Reply:
> > It's an executor instead?
> 
> It is an executor behind the scenes, see this please: [#508 (comment)](https://github.com/hazelcast/hz-docs/pull/508#discussion_r951240867)

Got it. I think we should mention it here in that case.